go/login/server/server.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package server implements GLOME-login server framework.
+package server
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"../../login" // Modify when it is pushed
+)
+
+const (
+	// MaxResponseSize is the maximum size in characaters of the response token
+	MaxResponseSize = 44 // 32 bytes base64 encoded
+)
+
+var (
+	// ErrInvalidResponseLen denotes that response length provided is invalid
+	ErrInvalidResponseLen = fmt.Errorf("invalid response length provided")
+)
```

Comment:
That's an anti-pattern here as well. Unless you need to check for the error, consider just returning `fmt.Errorf("invalid response length provided: got %v, want %v", ...)` or something like that at the site where you currently return this.

Reply:
Changed to a struct 
```
type ErrInvalidResponseLen struct {                                                                                     
    ResponseLen uint8                                                                                                   
}
```
That implements Error interface.